roles/openshift_health_checker/openshift_checks/docker_storage.py

```diff
@@ -0,0 +1,181 @@
+            return {"failed": True, "changed": changed,
+                    "msg": "Docker API query missing info:\n{}".format(json.dumps(info))}
+        info = info["info"]
+
+        # check if the storage driver we saw is valid
+        driver = info.get("Driver", "[NONE]")
+        if driver not in self.storage_drivers:
+            msg = (
+                "Detected unsupported Docker storage driver '{driver}'.\n"
+                "Supported storage drivers are: {drivers}"
+            ).format(driver=driver, drivers=', '.join(self.storage_drivers))
+            return {"failed": True, "changed": changed, "msg": msg}
+
+        # driver status info is a list of tuples; convert to dict and validate based on driver
+        driver_status = {item[0]: item[1] for item in info.get("DriverStatus", [])}
+        if driver == "devicemapper":
+            if self._is_dm_using_loopback_device(driver_status):
+                msg = (
+                    "Use of loopback devices with the Docker devicemapper storage driver\n"
+                    "(the default storage configuration) is unsupported in production.\n"
+                    "Please use docker-storage-setup to configure a backing storage volume.\n"
+                    "See http://red.ht/2rNperO for further information."
+                )
+                return {"failed": True, "changed": changed, "msg": msg}
+            result = self._check_dm_usage(driver_status, task_vars)
+            result["changed"] = changed
+            return result
+
+        # TODO(lmeyer): determine how to check usage for overlay2
+
+        return {"changed": changed}
+
+    @staticmethod
+    def _is_dm_using_loopback_device(driver_status):
+        return bool(driver_status.get("Data loop file"))
+
+    def _check_dm_usage(self, driver_status, task_vars):
+        """
+        For background: under thinpool-backed devicemapper, docker-storage-setup creates
+        an LVM2 LV as a backing storage pool. The LV is "thin" because it does not use
+        all available storage from its VG, instead expanding as needed; so in considering
+        available space, it is necessary to include space remaining in the VG (which is
+        unfortunately not reported by the storage driver in docker info).
+        Usage of the LV is divided into space allocated to data and metadata (which also
+        expand automatically), either of which could run out of space while the other is
+        fine; so both must be checked.
+        """
+        vals = dict(
+            vg_free=self._get_vg_free(driver_status.get("Pool Name"), task_vars),
+            data_used=driver_status.get("Data Space Used"),
+            data_total=driver_status.get("Data Space Total"),
+            metadata_used=driver_status.get("Metadata Space Used"),
+            metadata_total=driver_status.get("Metadata Space Total"),
+        )
+
+        for key, value in vals.copy().items():
```

Comment:
That's correct...

On a second look, I wonder why we mutate `vals`, and why do we add a key with the `_bytes` prefix? I couldn't find anywhere that actually reads back that key?!